encodings/dict/src/compress.rs

```diff
@@ -20,22 +21,66 @@ use crate::DictArray;
 /// Statically assigned code for a null value.
 pub const NULL_CODE: u64 = 0;
 
-#[derive(Debug)]
-struct Value<T>(T);
+mod private {
```

Comment:
this is trick so i can use Value as a type boundary on PrimitiveDictBuilder without actually leaking Value to the external pub api